src/test/java/org/threeten/extra/TestInterval.java

```diff
@@ -267,6 +269,39 @@ public void test_parse_CharSequence(String input, Instant start, Instant end) {
         assertEquals(end, test.getEnd());
     }
 
+    @ParameterizedTest
+    @CsvSource({
+        "-1000000000-01-31T00:00:00Z/P1M, -1000000000-01-31T00:00:00Z, -1000000000-02-29T00:00:00Z",
+        "-1000000000-01-31T00:00:00Z/P2M, -1000000000-01-31T00:00:00Z, -1000000000-03-31T00:00:00Z",
+        "P5M/+1000000000-12-31T23:59:59.999999999Z, +1000000000-07-31T23:59:59.999999999Z, +1000000000-12-31T23:59:59.999999999Z",
+        "P10M/+1000000000-12-31T23:59:59.999999999Z, +1000000000-02-29T23:59:59.999999999Z, +1000000000-12-31T23:59:59.999999999Z",
+    })
+    public void data_parse_outside_bounds(String interval, String start, String end) {
+        assertEquals(Interval.of(Instant.parse(start), Instant.parse(end)), Interval.parse(interval));
+    }
+
+    @Disabled("not implemented yet")
```

Comment:
Does this indicate that the PR is not ready yet?

Reply:
I removed "yet". This PR doesn't fix, but also doesn't break anything related to these cases. These 2 disabled tests are here to make others aware of implementation limitations that are present now and were present before, but are / were not so obvious at first sight.

> There are still some cases that this rewrite doesn't fix. All of them involve duration. Please, see `org.threeten.extra.TestInterval.data_parse_crossing_bounds(String, String, String)` (intervals starting within date-time bounds, with duration making them end outside of it) and `org.threeten.extra.TestInterval.data_parse_crossing_bounds_twice(String, String, String)` (intervals starting before OffsetDateTime.MIN and ending after `OffsetDateTime.MAX` because of duration). Maybe someone will be interested in making these tests pass - for now, they are disabled.

I've tried to implement such cases for a while, but code quickly becomes messy. Boundary between `Interval` and `OffsetDateTime` is hard to cross. `Instant.MIN.plusDays(1)` is relatively easy. 400 years forward, 1 day forward, 400 years backward, done. On the other hand, something like `OffsetLocalDate.MIN.plusDays(DAYS.between(LocalDate.MIN, LocalDate.MAX) + 1)` is _much_ harder to implement as _every_ shift will always end up outside `OffsetDateTime.MIN/MAX` bounds. 

I've also tried to experiment with `DateTimeFormatter.parseUnresolved`, but failed to find that helpful.

It seems that some `ExtendedOffsetDateTime` (using `ExtendedYear` internally (?), allowing values outside `Year.MIN/MAX`) + reimplementing all `TemporalAmount`-related calculations from scratch would be needed. That's too much for me.

To summarize: please consider this PR as ready.